integration/spark-common/src/main/scala/org/apache/carbondata/spark/rdd/CarbonScanRDD.scala

```diff
@@ -150,12 +153,25 @@ class CarbonScanRDD[V: ClassTag](
     val attemptContext = new TaskAttemptContextImpl(new Configuration(), attemptId)
     val format = prepareInputFormatForExecutor(attemptContext.getConfiguration)
     val inputSplit = split.asInstanceOf[CarbonSparkPartition].split.value
-    val reader = format.createRecordReader(inputSplit, attemptContext)
+    val model = format.getQueryModel(inputSplit, attemptContext)
+    val reader = {
+      if (vectorReader) {
+        val carbonRecordReader = createVectorizedCarbonRecordReader(model)
+        if (carbonRecordReader == null) {
+          new CarbonRecordReader(model, format.getReadSupportClass(attemptContext.getConfiguration))
+        } else {
+          carbonRecordReader
+        }
+      } else {
+        new CarbonRecordReader(model, format.getReadSupportClass(attemptContext.getConfiguration))
```

Comment:
should not new CarbonRecordReader directly, can we choose:
option 1: create two InputFormat, one for batch and another for non-batch
option 2: one InputFormat, create RecordReader according to configuration.
I prefer option 2.